### PR TITLE
LibC: Make h_errno thread-local.

### PR DESCRIPTION
**Pull request by cocateh**
This PR makes `h_errno` a thread-local variable, which it should be as per POSIX, along with adding some missing functions for describing its values.

### CODE REVIEW

**Comment by timschumi** [comment]
Are the `const_cast<char*>` in `hstrerror()` actually needed? If I remember correctly, string literals can be returned just fine in a function that returns `char const*`.

**Comment by cocateh** [comment]
Maybe, but I wanted it to be similar to our `strerror()` which does explicitly `const_cast`

**Comment by timschumi** [comment]
`strerror()` only does this because it returns a non-const `char*`, while mandating that the application should not modify the returned string. `hstrerror` doesn't need to do this, and therefore shouldn't use that workaround in the first place.

**Comment by cocateh** [comment]
Oh right, I didn't notice that `strerror()` is not `const`. Will change it.

File: Userland/Libraries/LibC/netdb.cpp [hunk 1]
```diff
@@ -825,4 +829,32 @@ int getnameinfo(const struct sockaddr* __restrict addr, socklen_t addrlen, char*
...
+        break;
+    case NO_RECOVERY:
+        return "A nonrecoverable name server error occurred";
+        break;
+    case TRY_AGAIN:
+        return "A temporary error occurred on an authoritative name server. Try again later.";
+        break;
+    default:
+        return "Unknown error.";
+        break;
```
Comment (Lubrsi) [inline Userland/Libraries/LibC/netdb.cpp:857]: The `break`s here aren't required (since it `return`s before it can ever `break`)

Comment (cocateh) [reply Userland/Libraries/LibC/netdb.cpp:857]: I will remove them.

File: Userland/Libraries/LibC/netdb.cpp [hunk 2]
```diff
@@ -829,4 +829,27 @@ int getnameinfo(const struct sockaddr* __restrict addr, socklen_t addrlen, char*
 
     return 0;
 }
+
+void herror(char const* s)
+{
+    int saved_herrno = h_errno;
+    dbgln("herror(): {}: {}", s, hstrerror(saved_herrno));
+    warnln("{}: {}", s, hstrerror(saved_herrno));
```
Comment (awesomekling) [inline Userland/Libraries/LibC/netdb.cpp:837]: Should be make sure to preserve `h_errno` s so that calling `herror()` doesn't clobber it?

Comment (cocateh) [reply Userland/Libraries/LibC/netdb.cpp:837]: Actually, it is unnecessary. Will fix.

**Review by kleinesfilmroellchen — APPROVED** [review]

**Pull request merged**